apps/vast/vast-bridge.py

```diff
@@ -97,17 +152,48 @@ async def receive_intel(cmd, vast_endpoint, pub_endpoint, topic):
             except Exception as e:
                 logger.error(f"Error decoding message: {e}")
                 continue
-            operation = intel.get("operation", None)
-            intel.pop("operation", None)
-            if operation == "ADD":
+            await intel_queue.put(intel)
+        else:
+            await asyncio.sleep(0.05)  # free event loop for other tasks
+
+
+async def match_intel(cmd, vast_endpoint, intel_queue, sightings_queue, retro_match):
+    """
+    Reads from the intel_queue and matches all IoCs, either via VAST's
+    live-matching (pro feature) or retro-matching.
+    @param cmd The vast binary command to use with PyVAST
+    @param vast_endpoint The endpoint of a running vast node
+    @param intel_queue The queue to read new IoCs from
+    @param sightings_queue The queue to put new sightings into
+    @param retro_match Boolean flag to use retro-matching over live-matching
+    """
+    global logger
+    vast = VAST(binary=cmd, endpoint=vast_endpoint)
+    while True:
+        intel = await intel_queue.get()
+        operation = intel.pop("operation", None)
+        if operation == "ADD":
+            if retro_match:
+                query = ioc_to_query(intel)
+                if not query:
+                    continue
+                proc = await vast.export().json(query).exec()
+                while not proc.stdout.at_eof():
+                    line = (await proc.stdout.readline()).decode().rstrip()
```

Comment:
`line` suggests that it's a single line. But I'd imagine a query can yield multiple. Where is the loop-over-the-lines logic implemented?

Reply:
This is the loop (line 181) - `while` process is not at `eof`, do `readline()` ...